lib/provider/azure/sanitize-headers.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const setCookieVariations = require('./set-cookie.json').variations;
+
+module.exports = function sanitizeHeaders(headers) {
+  return Object.keys(headers).reduce((memo, key) => {
+      const value = headers[key];
+
+      if (Array.isArray(value)) {
+        if (key.toLowerCase() === 'set-cookie') {
+          value.forEach((cookie, i) => {
+            memo[setCookieVariations[i]] = cookie;
+          });
+        } else {
+          memo[key] = value.join(', ');
+        }
+      } else {
+        memo[key] = value == null ? '' : value.toString();
+      }
+
+      return memo;
+    }, {});
+};
```

Comment:
@ncipollina nice workaround for the bug with multiple `set-cookie` headers. The issue seems to be somewhat fixed in https://github.com/Azure/azure-functions-host/issues/2737#issuecomment-568037718. 
Although it is not ideal as you can't just use multiple `set-cookie` headers, you need to use the `cookies` property of a response..
But maybe it would be a cleaner solution.